libcodechecker/plist.py

```diff
@@ -0,0 +1,332 @@
+    # -e core -d core.uninitialized -e core.uninitialized.Assign
+    # We must support having multiple '-e' and '-d' options and the order
+    # specified must be kept when the list of checkers are assembled for Clang.
+
+    def __init__(self, option_strings, dest, nargs=None, **kwargs):
+        if nargs is not None:
+            raise ValueError("nargs not allowed")
+        super(OrderedCheckersAction, self).__init__(option_strings, dest,
+                                                    **kwargs)
+
+    def __call__(self, parser, namespace, value, option_string=None):
+
+        if 'ordered_checkers' not in namespace:
+            namespace.ordered_checkers = []
+        ordered_checkers = namespace.ordered_checkers
+        ordered_checkers.append((value, self.dest == 'enable'))
+
+        namespace.ordered_checkers = ordered_checkers
+
+
+class DeprecatedOptionAction(argparse.Action):
+    """
+    Deprecated argument action.
+    """
+
+    def __init__(self,
+                 option_strings,
+                 dest,
+                 nargs=None,
+                 const=None,
+                 default=None,
+                 type=None,
+                 choices=None,
+                 required=False,
+                 help=None,
+                 metavar=None):
+        super(DeprecatedOptionAction, self). \
+            __init__(option_strings,
+                     dest,
+                     const='deprecated_option',
+                     default=argparse.SUPPRESS,
+                     type=None,
+                     choices=None,
+                     required=False,
+                     help="(Usage of this argument is DEPRECATED and has no "
+                          "effect!)",
+                     metavar='')
+
+    def __call__(self, parser, namespace, value=None, option_string=None):
+        LOG.warning("Deprecated command line option used: '" +
+                    option_string + "'")
+
+
+def get_argparser_ctor_args():
+    """
+    This method returns a dict containing the kwargs for constructing an
+    argparse.ArgumentParser (either directly or as a subparser).
+    """
+
+    return {
+        'prog': 'CodeChecker plist',
+        'formatter_class': argparse.ArgumentDefaultsHelpFormatter,
+
+        # Description is shown when the command's help is queried directly
+        'description': "Parse plist files in the given directory and "
+                       "store the defects found therein to the database "
+                       "or print to the standard output.",
+
+        # Help is shown when the "parent" CodeChecker command lists the
+        # individual subcommands.
+        'help': "Use plist files in a given directory to pretty-print or "
+                "store the results."
+    }
+
+
+def add_arguments_to_parser(parser):
+    """
+    Add the subcommand's arguments to the given argparse.ArgumentParser.
+    """
+
+    # TODO: --name is not required in 'store'.
```

Comment:
Same answer as before. `--name` **is** an argument for `store`, with `required=False`. But to keep previous invocation consistent, `--name` is `reuqired=True` in `plist` and `check`. (Though in case of `plist` it's a mistake, see last two paragraphs of the patch description.)